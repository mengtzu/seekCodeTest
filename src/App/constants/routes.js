const root = '/';
const adPacks = '/adPacks';
const checkout = '/checkout';
const confirmation = '/confirmation';
const logout = '/Login/Logout';

export const routes = {
    root,
    adPacks,
    checkout,
    confirmation,
    logout
};

export default routes;
