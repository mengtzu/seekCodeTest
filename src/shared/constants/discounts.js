export const DISCOUNT_TYPE_FLAT_RATE = 'flatRate';
export const DISCOUNT_TYPE_THRESHOLD = 'threshold';
export const DISCOUNT_TYPE_X_FOR_Y = 'xfory';
