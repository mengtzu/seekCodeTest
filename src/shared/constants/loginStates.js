export const AUTH_STATUS_UNAUTHENTICATED = 'pending'; //'unauthenticated' exposes the real SEEK login, don't want that!
export const AUTH_STATUS_AUTHENTICATED = 'authenticated';
