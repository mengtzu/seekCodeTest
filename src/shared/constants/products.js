export const CLASSIC = 'classic';
export const STANDOUT = 'standout';
export const PREMIUM = 'premium';
