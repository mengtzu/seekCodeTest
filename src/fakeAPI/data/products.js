import { CLASSIC, STANDOUT, PREMIUM } from '../../shared/constants/products';

let productsData = {};

//Bracket notation isn't necessarily preferred but here it lets us avoid having to iterate over this object in various places

productsData[CLASSIC] = {
    id: CLASSIC,
    basePrice: 269.99
};

productsData[STANDOUT] = {
    id: STANDOUT,
    basePrice: 322.99
};

productsData[PREMIUM] = {
    id: PREMIUM,
    basePrice: 394.99
}

export default productsData;