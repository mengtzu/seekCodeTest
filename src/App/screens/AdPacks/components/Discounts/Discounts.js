import React from 'react';
import { connect } from 'react-redux';
import { Text } from 'seek-style-guide/react';

import { DISCOUNT_TYPE_FLAT_RATE, DISCOUNT_TYPE_X_FOR_Y, DISCOUNT_TYPE_THRESHOLD } from '../../../../../shared/constants/discounts';

const mapStateToProps = (state) => {
    return {
        discounts: state.advertiser.discounts
    }
};

export const Discounts = ({ adType, discounts }) => {
    const discountsForDisplay = discounts.map((discount, index) => {
        if (discount.ad === adType) {
            switch (discount.type) {
                case DISCOUNT_TYPE_FLAT_RATE:
                    return (
                        <Text positive key={index}>Your price: ${discount.price}</Text>
                    );
                    break;
                case DISCOUNT_TYPE_THRESHOLD:
                    return (
                        <Text positive key={index}>Your price: ${discount.price} when you buy {discount.threshold} or more</Text>
                    );
                    break;
                case DISCOUNT_TYPE_X_FOR_Y:
                    return (
                        <Text positive key={index}>Get {discount.x} ads for the price of {discount.y}</Text>
                    );
                    break;
            }
        }
    });

    if (discountsForDisplay.length) {
        return (
            <div>{discountsForDisplay}</div>
        )
    } else {
        return null;
    }
};

export default connect(mapStateToProps)(Discounts);
