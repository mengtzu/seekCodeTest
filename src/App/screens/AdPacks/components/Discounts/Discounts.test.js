import React from 'react';
import { Discounts } from './Discounts';
import renderer from 'react-test-renderer';

import { DISCOUNT_TYPE_FLAT_RATE, DISCOUNT_TYPE_X_FOR_Y, DISCOUNT_TYPE_THRESHOLD } from '../../../../../shared/constants/discounts';
import { CLASSIC, STANDOUT, PREMIUM } from '../../../../../shared/constants/products';

const mockDiscounts = [{
    type: DISCOUNT_TYPE_FLAT_RATE,
    ad: CLASSIC,
    price: 199.99
}, {
    type: DISCOUNT_TYPE_THRESHOLD,
    ad: CLASSIC,
    price: 22,
    threshold: 4
}, {
    type: DISCOUNT_TYPE_X_FOR_Y,
    ad: CLASSIC,
    x: 5,
    y: 1
}];

it('renders correctly with valid discounts', () => {
    const tree = renderer.create(
        <Discounts adType={CLASSIC} discounts={mockDiscounts} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with no valid discounts', () => {
    const tree = renderer.create(
        <Discounts adType={STANDOUT} discounts={mockDiscounts} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with no discounts', () => {
    const tree = renderer.create(
        <Discounts adType={PREMIUM} discounts={[]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});


