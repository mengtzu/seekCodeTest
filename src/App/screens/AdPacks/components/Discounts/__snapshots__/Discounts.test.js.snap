// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no discounts 1`] = `null`;

exports[`renders correctly with no valid discounts 1`] = `<div />`;

exports[`renders correctly with valid discounts 1`] = `
<div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Your price: $
      199.99
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Your price: $
      22
       when you buy 
      4
       or more
    </span>
  </div>
  <div
    className="root"
  >
    <span
      className=""
    >
      Get 
      5
       ads for the price of 
      1
    </span>
  </div>
</div>
`;
